language/stdlib/modules/Roles.move

```diff
@@ -0,0 +1,265 @@
+address 0x1 {
+/// This module describes two things:
+/// 1. The relationship between roles, e.g. Role_A can creates accounts of Role_B
+/// 2. The granting of privileges to an account with a specific role
+/// It is important to note here that this module _does not_ describe the
+/// privileges that a specific role can have. This is a property of each of
+/// the modules that declares a privilege.
```

Comment:
Might be useful to add a quick terminology primer somewhere here:
- `Privilege<T>`: label on an account signifying that the account can do `T`
- `Capability<T>`: object representing the authority to do `T`. Requested by procedures of the module that implements `T`

This is a question for later, but I'm wondering if we want a term to distinguish stuck privileges with temporary capabilities from things like `WithdrawCapability` (which are not stuck).

Reply:
Good point on terminology, I'll add it in.

In terms of unstuck capabilities (`WithdrawCapability`, `MintCapability<T>` etc), I think having a good way of distinguishing those is important. I think if we could build a unified model for representing these like this does for roles/"stuck" privileges, that would be very nice indeed...